dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/AppClassLoader.java

```diff
@@ -319,42 +329,60 @@ boolean removeTransformer(ClassFileTransformer transformer) {
         // types of exceptions or quitting out of the transformation code which will result in the class not being
         // transformed anyway.
         byte[] bytes;
-        if (byteResourceInformation.foundInClassCache()) {
-            bytes = byteResourceInformation.getBytes(); 
-            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                Tr.debug(tc, "Not transforming class " + name + " because it was found in the shared classes cache.");
+        if (toTransform.foundInClassCache()) {
+            if (systemTransformers.isEmpty()) {
+                bytes = toTransform.getBytes(); 
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "Not transforming class " + name + " because it was found in the shared classes cache.");
+                }
+            } else {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "Attempt to transform " + name + " from shared class cache because we have global tranformers");
```

Comment:
No, I had started with a variable named `global` but then decided to rename it to `system` because that seemed a better name.  I'll update the debug message.